sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -1214,6 +1215,13 @@ class CastSuite extends CastSuiteBase {
       checkEvaluation(Cast(Literal(134.12), DecimalType(3, 2)), null)
     }
   }
+
+  test("SPARK-30008: collect_list/collect_set can cast to ArrayType not containsNull") {
+    val list = CollectList(Literal(1))
+    assert(Cast.canCast(list.dataType, ArrayType(IntegerType, false)))
+    val set = CollectSet(Literal(1))
+    assert(Cast.canCast(set.dataType, ArrayType(StringType, false)))
+  }
```

Comment:
And, the following PR description is correct?
```
Fix bug. i.e. casting collect_list(a) to ArrayType(_, false) will fail.
```

Reply:
Thanks, should be  `Fix bug. i.e. casting collect_list(a) to ArrayType(_, false) will fail **before this fix**.`